packages/frontend-api/src/Model/Resolver/Article/ArticlesResolver.php

```diff
@@ -0,0 +1,126 @@
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Overblog\GraphQLBundle\Definition\Resolver\AliasedInterface;
+use Overblog\GraphQLBundle\Definition\Resolver\ResolverInterface;
+use Overblog\GraphQLBundle\Relay\Connection\ConnectionBuilder;
+use Overblog\GraphQLBundle\Relay\Connection\Paginator;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Article\ArticleFacade;
+
+class ArticlesResolver implements ResolverInterface, AliasedInterface
+{
+    protected const DEFAULT_FIRST_LIMIT = 10;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Article\ArticleFacade
+     */
+    protected $articleFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Overblog\GraphQLBundle\Relay\Connection\ConnectionBuilder
+     */
+    protected $connectionBuilder;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Article\ArticleFacade $articleFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(ArticleFacade $articleFacade, Domain $domain)
+    {
+        $this->articleFacade = $articleFacade;
+        $this->domain = $domain;
+        $this->connectionBuilder = new ConnectionBuilder();
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @param string|null $placement
+     * @return \Overblog\GraphQLBundle\Relay\Connection\ConnectionInterface|object
+     */
+    public function resolve(Argument $argument, ?string $placement)
+    {
+        $this->setDefaultFirstOffsetIfNecessary($argument);
+        $domainId = $this->domain->getId();
+
+        $paginator = new Paginator(function ($offset, $limit) use ($domainId, $placement) {
+            return $this->getArticlesLimited($domainId, $limit, $offset, $placement);
+        });
+
+        return $paginator->auto($argument, $this->getArticlesCount($domainId, $placement));
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     */
+    protected function setDefaultFirstOffsetIfNecessary(Argument $argument): void
+    {
+        if ($argument->offsetExists('first') === false
+            && $argument->offsetExists('last') === false
+        ) {
+            $argument->offsetSet('first', static::DEFAULT_FIRST_LIMIT);
+        }
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getAliases(): array
+    {
+        return [
+            'resolve' => 'articles',
+        ];
+    }
+
+    /**
+     * @param int $domainId
+     * @param int $limit
+     * @param int $offset
+     * @param string|null $placement
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article[]
+     */
+    protected function getArticlesLimited(
+        int $domainId,
+        int $limit,
+        int $offset,
+        ?string $placement
+    ): array {
+        if ($placement !== null) {
```

Comment:
In case we are using guard, I would prefer if condition would be positive ($placement === null) if possible

Reply:
I would rather keep this one.

IMO it has nothing to do with guard statement because any of both states are not invalid and it is hard to tell what is the expected path (If the `$placement = null` would result in exception I would agree with you).

Reply:
fixed even in `::getArticlesCount()`